Modules/CLI/ResampleDTIVolume/itkDiffusionTensor3DMatrix3x3Transform.h

```diff
@@ -96,7 +100,11 @@ class DiffusionTensor3DMatrix3x3Transform :
   VectorType                  m_Translation;
   VectorType                  m_Offset;
   PointType                   m_Center;
-  MutexLock::Pointer          m_Lock;
+#if ITK_VERSION_MAJOR >= 5
```

Comment:
Should we just keep the case of with ` #if ITK_VERSION_MAJOR >= 5 ` ?

Reply:
Sorry I don't understand what we should keep

Reply:
Instead of (re-)introducing the support for both ITKv5 and ITKv4, I suggest we simply update the file to use the ITKv5 convention.

In this case, it means that instead of having:

```
#if ITK_VERSION_MAJOR >= 5
  std::mutex                  m_Lock;
#else
  MutexLock                   m_Lock;
#endif
```

we would use only 

```
std::mutex                  m_Lock;
```

Is this PR related to https://discourse.slicer.org/t/slicerrt-does-not-build-due-to-missing-itk-remote-module/6214 ? 

And is the template code only used in extension ?

Reply:
No I got this build error when building a SlicerCustomApp. Not sure why it doesn't fail in a normal Slicer build.
I can update the PR to only use ITKv5, but then what happens if the user configures Slicer with Slicer_ITK_VERSION_MAJOR=4?

Reply:
Since we removed the support for ITKv4, it should be a non issue. See https://discourse.slicer.org/t/2019-03-05-hangout/6007/2?u=jcfr

I suggest to update the custom app following https://www.slicer.org/wiki/Documentation/Nightly/Developers/Tutorials/MigrationGuide#Transition_from_ITK4_to_ITK5

Reply:
OK.
I don't understand why the custom app build fails, it uses neither ResampleDTIVolume, or any ITK directly. Or rather, I don't understand why Slicer build was successful...